### PR TITLE
Discord output

### PR DESCRIPTION
**Pull request by nibalizer**
Added a discord output source. Ready for review!

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ func getConfig() *types.Configuration {
 	v.SetDefault("Datadog.APIKey", "")
 	v.SetDefault("Datadog.Host", "https://api.datadoghq.com")
 	v.SetDefault("Datadog.MinimumPriority", "")
+	v.SetDefault("Discord.WebhookURL", "")
```
Comment (Issif) [inline config.go:40]:
```suggestion
	v.SetDefault("Discord.WebhookURL", "")
	v.SetDefault("Discord.MinimumPriority", "")
```

File: outputs/discord.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/falcosecurity/falcosidekick/types"
+)
+
+type discordPayload struct {
+	Content    string `json:"content"`
+	Avatar_url string `json:"avatar_url,omitempty"`
+}
+
+func newDiscordPayload(falcopayload types.FalcoPayload, config *types.Configuration) discordPayload {
+	iconURL := "https://raw.githubusercontent.com/falcosecurity/falcosidekick/master/imgs/falcosidekick.png"
```
Comment (Issif) [inline outputs/discord.go:13]: It should be a configurable option like for `Slack`.

File: outputs/discord.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"github.com/falcosecurity/falcosidekick/types"
+)
+
+type discordPayload struct {
+	Content    string `json:"content"`
+	Avatar_url string `json:"avatar_url,omitempty"`
+}
+
```
Comment (Issif) [inline outputs/discord.go:11]: Is it possible to enhance that payload to get a formatted message with a color according to priority and embeds?

See : https://discord.com/developers/docs/resources/webhook

File: outputs/discord.go (hunk 2 above)
Comment (nibalizer) [reply outputs/discord.go:13]: ARe you looking for a binary "custom avatar" = yes/no or a configurable avatar?

File: outputs/discord.go (hunk 3 above)
Comment (nibalizer) [reply outputs/discord.go:11]: I'll probably add *some* of the configurability of the slack hook but I don't want to let "perfect be the enemy of the good" if that makes sense.

File: outputs/discord.go (hunk 2 above)
Comment (Issif) [reply outputs/discord.go:13]: A configurable avatar, and this as default.

File: outputs/discord.go (hunk 3 above)
Comment (Issif) [reply outputs/discord.go:11]: I agree with you, but it's important to me to have nice output, it was my main purpose at start.

**Comment by Issif** [comment]
@nibalizer What about my proposal on Slack to format the output with colors, etc, like for Slack and others?

**Comment by nibalizer** [comment]
Yes I think we should do it. I can work on this a bit today and will. Sorry for the delay.

File: outputs/discord.go (hunk 3 above)
Comment (nibalizer) [reply outputs/discord.go:11]: So it doesn't look like discord has the same priority hooks that Slack has. There are some hacks (and I can't emphasize enough that they hare hacks) at [this link](https://www.writebots.com/discord-text-formatting/). Red seems easy and not-terrible if you'd like me to add that.

**Comment by nibalizer** [comment]
@Issif please see my note above, I don't see a good way to do colors for messages in discord.

File: outputs/discord.go (hunk 3 above)
Comment (Issif) [reply outputs/discord.go:11]: I thought it was possible with embed objects : https://discord.com/developers/docs/resources/channel#embed-object

Here a result : https://birdie0.github.io/discord-webhooks-guide/discord_webhook.html

Comment (nibalizer) [reply outputs/discord.go:11]: Oh Cool! Thanks for the example. I messed with `embed` for a minute but couldn't get it working. I'll try to incorporate.

**Comment by nibalizer** [comment]
Color is working, need to write some more code before it'll work/look right

image

**Comment by nibalizer** [comment]
Thanks for the tip @Issif

**Comment by Issif** [comment]
Nice ! Can you display `fields` as seperated? One again, like in `Slack` or `Teams`.

**Comment by nibalizer** [comment]
Sorry team! I'll try to get back to this as soon as I can!

**Comment by Issif** [comment]
@nibalizer no problem, I'm really busy on my side

File: README.md [hunk 4]
```diff
@@ -407,6 +416,10 @@ time                akey    bkey    ckey    priority rule      value
 
 ![opsgenie example](https://github.com/falcosecurity/falcosidekick/raw/master/imgs/opsgenie.png)
 
+### Discord
+
+![discord example](imgs/discord_example.png)
```
Comment (Issif) [inline README.md:421]:
```suggestion
![discord example](https://github.com/falcosecurity/falcosidekick/raw/master/imgs/discord_example.png)
```

This is mandatory to use the full URL for having images in dockerhub page

Comment (nibalizer) [reply README.md:421]: Ah ok!

**Pull request merged**